tests/common.bash

```diff
@@ -542,13 +542,32 @@ function install_nydus() {
 
 function install_nydus_snapshotter() {
 	version="${1}"
-
-	project="containerd/nydus-snapshotter"
-	tarball_name="nydus-snapshotter-${version}-$(${repo_root_dir}/tests/kata-arch.sh).tgz"
-
-	download_github_project_tarball "${project}" "${version}" "${tarball_name}"
-	sudo tar xfz "${tarball_name}" -C /usr/local/bin --strip-components=1
-	rm -f "${tarball_name}"
+	shift
+	
+	if [ $# -lt 1 ]; then
+		project="containerd/nydus-snapshotter"
+		tarball_name="nydus-snapshotter-${version}-$(${repo_root_dir}/tests/kata-arch.sh).tgz"
+
+		download_github_project_tarball "${project}" "${version}" "${tarball_name}"
+		sudo tar xfz "${tarball_name}" -C /usr/local/bin --strip-components=1
+		rm -f "${tarball_name}"
+	else
+		nydus_snaphshotter_dir="$(mktemp -d -t nydus_snaphshotter.XXXXXXXXXX)/nydus-snapshotter"
+		nydus_snapshotter_config_dir="/etc/nydus-snapshotter"
+		git clone ${nydus_snapshotter_repo} "${nydus_snaphshotter_dir}" || true
```

Comment:
Thanks @stevenhorsman  . I also hope we can use the release binary rather than built it. I will collaborate with @jiangliu  to reach out to the nydus-snapshotter team and move this forward.